src/allmydata/test/web/test_root.py

```diff
@@ -57,8 +56,17 @@ def test_invalid(self):
         A (trivially) invalid capbility is an error
         """
         self.request.args[b"uri"] = [b"not a capability"]
-        with self.assertRaises(WebError):
-            self.res.render_GET(self.request)
+        response_body = self.res.render_GET(self.request)
+
+        self.assertIn(
+            "<title>400 - Error</title>", response_body,
```

Comment:
A lot of the tests are already using `BeautifulSoup` to confirm structure of HTML responses -- that could make these tests more stable.